homeassistant/components/matter/climate.py

```diff
@@ -0,0 +1,327 @@
+
+        current_mode = target_hvac_mode or self.hvac_mode
+        command = None
+        if current_mode in (HVACMode.HEAT, HVACMode.COOL):
+            # when current mode is either heat or cool, the temperature arg must be provided.
+            temperature: float | None = kwargs.get(ATTR_TEMPERATURE)
+            if temperature is None:
+                raise ValueError("Temperature must be provided")
+            if self.target_temperature is None:
+                raise ValueError("Current target_temperature should not be None")
+            command = self._create_optional_setpoint_command(
+                clusters.Thermostat.Enums.SetpointAdjustMode.kCool
+                if current_mode == HVACMode.COOL
+                else clusters.Thermostat.Enums.SetpointAdjustMode.kHeat,
+                temperature,
+                self.target_temperature,
+            )
+        elif current_mode == HVACMode.HEAT_COOL:
+            temperature_low: float | None = kwargs.get(ATTR_TARGET_TEMP_LOW)
+            temperature_high: float | None = kwargs.get(ATTR_TARGET_TEMP_HIGH)
+            if temperature_low is None or temperature_high is None:
+                raise ValueError(
+                    "temperature_low and temperature_high must be provided"
+                )
+            if (
+                self.target_temperature_low is None
+                or self.target_temperature_high is None
+            ):
+                raise ValueError(
+                    "current target_temperature_low and target_temperature_high should not be None"
+                )
+            # due to ha send both high and low temperature, we need to check which one is changed
+            command = self._create_optional_setpoint_command(
+                clusters.Thermostat.Enums.SetpointAdjustMode.kHeat,
+                temperature_low,
+                self.target_temperature_low,
+            )
+            if command is None:
+                command = self._create_optional_setpoint_command(
+                    clusters.Thermostat.Enums.SetpointAdjustMode.kCool,
+                    temperature_high,
+                    self.target_temperature_high,
+                )
+        if command:
+            await self.matter_client.send_device_command(
+                node_id=self._endpoint.node.node_id,
+                endpoint_id=self._endpoint.endpoint_id,
+                command=command,
+            )
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set new target hvac mode."""
+        system_mode_path = create_attribute_path_from_attribute(
+            endpoint_id=self._endpoint.endpoint_id,
+            attribute=clusters.Thermostat.Attributes.SystemMode,
+        )
+        system_mode_value = HVAC_SYSTEM_MODE_MAP.get(hvac_mode)
+        if system_mode_value is None:
+            raise ValueError(f"Unsupported hvac mode {hvac_mode} in Matter")
+        await self.matter_client.write_attribute(
```

Comment:
How should we test the "write attribute"? I see that the current matter test cases are related to "send command"

Reply:
remaining uncovered three tests need to be able to test the write attribute before they can be added

Reply:
We can assert the calls to `matter_client.write_attribute` the same way we assert calls to `matter_client.send_device_command`.